src/data/data_representation_converter.cpp

```diff
@@ -0,0 +1,436 @@
+    {cudf::type_id::INT64,   duckdb::LogicalTypeId::BIGINT},
+
+    // Unsigned Integers
+    {cudf::type_id::UINT8,   duckdb::LogicalTypeId::UTINYINT},
+    {cudf::type_id::UINT16,  duckdb::LogicalTypeId::USMALLINT},
+    {cudf::type_id::UINT32,  duckdb::LogicalTypeId::UINTEGER},
+    {cudf::type_id::UINT64,  duckdb::LogicalTypeId::UBIGINT},
+
+    // Floating Point
+    {cudf::type_id::FLOAT32, duckdb::LogicalTypeId::FLOAT},
+    {cudf::type_id::FLOAT64, duckdb::LogicalTypeId::DOUBLE},
+
+    // Boolean & String
+    {cudf::type_id::BOOL8,   duckdb::LogicalTypeId::BOOLEAN},
+    {cudf::type_id::STRING,  duckdb::LogicalTypeId::VARCHAR},
+
+    // Timestamps
+    // cudf::TIMESTAMP_DAYS -> DuckDB DATE (days since epoch)
+    {cudf::type_id::TIMESTAMP_DAYS,         duckdb::LogicalTypeId::DATE},
+    {cudf::type_id::TIMESTAMP_SECONDS,      duckdb::LogicalTypeId::TIMESTAMP_SEC},
+    {cudf::type_id::TIMESTAMP_MILLISECONDS, duckdb::LogicalTypeId::TIMESTAMP_MS},
+    {cudf::type_id::TIMESTAMP_MICROSECONDS, duckdb::LogicalTypeId::TIMESTAMP}, // Default TIMESTAMP is us
+    {cudf::type_id::TIMESTAMP_NANOSECONDS,  duckdb::LogicalTypeId::TIMESTAMP_NS},
+};
+
+inline duckdb::LogicalTypeId cudf_type_to_duckdb_type(cudf::type_id id) {
+    auto it = CUDF_TO_DUCKDB_TYPES_MAP.find(id);
+    if (it != CUDF_TO_DUCKDB_TYPES_MAP.end()) {
+        return it->second;
+    }
+
+    return duckdb::LogicalTypeId::INVALID;
+}
+
+// Helper function used to determine the number of bytes needed to store a particular column
+size_t determine_column_bytes_needed(cudf::column_view& column_view, rmm::cuda_stream_view& stream) { 
+    size_t num_rows = column_view.size();
+
+    // First determine the bytes needed to hold the null mask
+    size_t null_mask_bytes = cudf::bitmask_allocation_size_bytes(num_rows); // By default aligns to 64 byte boundary
+
+    // Now determine the bytes needed to hold the actual data
+    size_t data_bytes = 0;
+    cudf::data_type col_type = column_view.type();
+    if(col_type.id() == cudf::type_id::STRING) { 
+        // For string get the number bytes for the offset and chars column. Also account for the space needed for the duckdb
+        // representation
+        cudf::strings_column_view strings_view(column_view);
+        size_t offset_bytes = num_rows * cudf::size_of(strings_view.offsets().type());
+        size_t chars_bytes = strings_view.chars_size(stream);
+        size_t duckdb_string_bytes = num_rows * sizeof(duckdb::string_t);
+        data_bytes = offset_bytes + chars_bytes + duckdb_string_bytes;
+    } else {
+        int32_t width = cudf::size_of(col_type);
+        data_bytes = width * num_rows;
+    }
+
+    return null_mask_bytes + data_bytes;
+}
+
+// Ensures that the location is aligned to the specificed size within the block
+inline void align_location(sirius::memory::result_table_location& location, size_t target_alignment, fixed_size_host_memory_resource::multiple_blocks_allocation& allocated_blocks) { 
```

Comment:
What is the purpose of this alignment? The multiple block allocations should not be aligned or have any constraints of this sort. The point is to densely pack things into them and leave alignment constraints to encodings that need to read and manipulate data. These representations are meant to hold bytes until they are ready to be converted to an encoding that is for processing.